src/tests/nativeaot/SmokeTests/Reflection/Reflection.cs

```diff
@@ -2700,6 +2701,31 @@ public static void Run()
         }
     }
 
+    // Regression test for https://github.com/dotnet/runtime/issues/111578
+    class TestGenericAttributesOnEnum
+    {
+        [AttributeUsage(AttributeTargets.Enum)]
+        class MyAttribute<T> : Attribute { }
+
+        [MyAttribute<int>]
+        enum MyEnum { A, B, C, D }
+
+        private static void GetEnumValues()
+        {
+            var values = Enum.GetValues<MyEnum>();
+            Assert.Equal(4, values.Length);
+            Assert.Equal(MyEnum.A, values[0]);
+            Assert.Equal(MyEnum.B, values[1]);
+            Assert.Equal(MyEnum.C, values[2]);
+            Assert.Equal(MyEnum.D, values[3]);
+        }
+
+        public static void Run()
+        {
+            GetEnumValues();
+        }
+    }
```

Comment:
The bug was in reading FlagsAttribute, it's a bit better to have a test that directly exercises something that needs it. GetValues doesn't need it, it just happens to run the codepath that reads it anyway.

```suggestion
    class TestGenericAttributesOnEnum
    {
        class MyAttribute<T> : Attribute { }

        [My<int>]
        enum MyEnum { A = 1, B = 2 }

        [MethodImpl(MethodImplOptions.NoInlining)]
        static object GetVal() => MyEnum.A | MyEnum.B;

        public static void Run()
        {
            if (GetVal().ToString() != "3")
                throw new Exception();
        }
    }
```